admin/starter/tests/unit/HealthTest.php

```diff
@@ -16,18 +18,35 @@ public function testIsDefinedAppPath()
 
 	public function testBaseUrlHasBeenSet()
 	{
-		$env = $config = false;
+		$validation = Services::validation();
+		$env = false;
 
-		// First check in .env
+		// Check the baseURL in .env
 		if (is_file(HOMEPATH . '.env'))
 		{
-			$env = (bool) preg_grep("/^app\.baseURL = './", file(HOMEPATH . '.env'));
+			$env = (bool) preg_grep('/^app\.baseURL = ./', file(HOMEPATH . '.env'));
+		}
+
+		if ($env)
+		{
+			// BaseURL in .env is a valid URL?
+			// phpunit.xml.dist sets app.baseURL in $_SERVER
+			// So if you set app.baseURL in .env, it takes precedence
+			$config = new Config\App();
+			$this->assertTrue(
+				$validation->check($config->baseURL, 'valid_url'),
+				'baseURL "' . $config->baseURL . '" in .env is not valid URL'
+			);
 		}
 
-		// Then check the actual config file
+		// Get the baseURL in app/Config/App.php
+		// Don't forget phpunit.xml.dist sets app.baseURL
```

Comment:
I think this is not needed. As `ConfigReader` does not call `Config\App`'s constructor, its properties would not be populated by environment values but only by the hardcoded values.

Reply:
@paulbalandan I fixed the comment. My intention is why we need to use ConfigReader.
I forgot about phpunit.xml.dist at the first time.